File: src/models/platform.go
package models

import "gorm.io/gorm"

type Platform struct {
	ID      int    `json:"id" gorm:"primaryKey"`
	Name    string `json:"name" gorm:"not null;size:64"`
	URL     string `json:"url" gorm:"size:64"`
	Country string `json:"country" gorm:"size:64"`
}

func GetPlatforms() (platforms []Platform, err error) {
	err = DB.Transaction(func(tx *gorm.DB) error {
		return tx.Find(&platforms).Error
	})
	if err != nil {
		return nil, err
	}
	return
}

func GetPlatformByID(platformID int) (platform *Platform, err error) {
	err = DB.Transaction(func(tx *gorm.DB) error {
		return tx.Take(&platform, platformID).Error
	})
	if err != nil {
		return nil, err
	}
	return
}

File: database-2023-autumn/GenerateData/src/main/java/pj/dbs/entity/Platform.java
package pj.dbs.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Platform {
    //主键：id
    //long id;     //auto
    String name;   //2-20
    String url;             //5-100
    String country;         //2-20
}
